### PR TITLE
feat: add data-dir support

### PR DESCRIPTION
**Pull request by FrankYang0529**
https://github.com/harvester/harvester/issues/6227

### CODE REVIEW

**Review by khushboo-rancher — APPROVED** [review]

File: docs/rancher/cloud-provider.md [hunk 1]
```diff
@@ -59,6 +59,11 @@ When spinning up an RKE2 cluster using the Harvester node driver, select the `Ha
 
   ![](/img/v1.2/rancher/rke2-cloud-provider.png)
 
+With Rancher version above 2.9.0, to customize folder for the cloud config data, you can modify `Data directory configuration path`.
```
Comment (jillian-maroket) [inline docs/rancher/cloud-provider.md:62]:
```suggestion
Starting with Rancher v2.9.0, you can configure a specific folder for cloud config data using the **Data directory configuration path** field.
```

Comment (jillian-maroket) [reply docs/rancher/cloud-provider.md:62]: We use backticks for code and bold for most GUI elements.

**Review by jillian-maroket — COMMENTED** [review]
I rewrote the sentence to make it flow better. PTAL

File: docs/rancher/cloud-provider.md (hunk 1 above)
Comment (FrankYang0529) [reply docs/rancher/cloud-provider.md]: Thanks for the review. Updated it.

**Review by jillian-maroket — APPROVED** [review]

**Pull request merged**